lib-object-oriented/line.js

```diff
@@ -1,34 +1,77 @@
 import Tag from './tag'
 
 export default class Line {
-  constructor( content ) {
+  constructor({ content, type }) {
     this.content = content
+    this.tagType = type
+    this.symbolCount = 0
+    this.getSymbolCount()
+    this.newLinesToSpaces()
+  }
+
+  newLinesToSpaces() {
+    const tagType = this.tagType,
+          content = this.content,
+          symbolCount = this.symbolCount
+
+    const findNewLine = element => {
+      return element.type === 'new_line'
+    }
+
+    if ( tagType !== '`' || ( tagType === '`' && symbolCount === 1 ) ) {
+      while ( content.findIndex( findNewLine ) > -1 ) {
+        const newLine = content.find( findNewLine )
+        if ( newLine ) {
+          newLine.type = ' '
+          newLine.value = ' '
+        }
+      }
+    }
+  }
+
+  getSymbolCount() {
+    for ( let index = 0; index < this.content.length; index++ ) {
+      const currentToken = this.content[ index ]
+      if ( currentToken.type.search( /([#>`])/g) !== -1 ) {
```

Comment:
This is a missed abstraction - what you're asking is if the `Token` object is a of a certain type, and this could be encapsulated in the `Token` object as a function:
```js
if( currentToken.isWhateverThisTestAsksFor() ) { // Again, not entirely certain what we're testing for
```